ANW 1002:  Map MARC 245$g to bulk dates on import

abbistani opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Fix for ANW-1002, MARC 245$g should import into a bulk date field

## Related JIRA Ticket or GitHub Issue
[ANW-1002](https://archivesspace.atlassian.net/browse/ANW-1002)

## How Has This Been Tested?
Tested with the Columbia example MARC record set https://github.com/ASpace-Metadata-Standards/marc-examples/tree/master/ColumbiaUniversityMARC

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have authority to submit this code.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

lorawoodford commented:
Hi @abbistani I think I'm seeing one problem/issue with the logic here (and that isn't accounted for in the sample set of MARC you've been testing against).

If there's *only* a 245$g, but no 245$f or 008, positions 7-14 (no matter the position 6), then I believe the import should still work and should result in one date record with the type of "bulk."  As this stands, a record of that type will fail to import and toss the following error:

```
The following errors were found:
	dates/0/expression : is required unless a begin or end date is given
	dates/0/begin : is required unless an expression or an end date is given
	dates/0/end : is required unless an expression or a begin date is given
```

This stems from the reliance on `expression` (which, in the scenario above is nil) on ln 778 of `backend/app/converters/lib/marcxml_base_map.rb`.  

Otherwise, all of the other examples provided in the sample MARC import as expected.

abbistani commented:
Got it.  The existing code created a blank date in those circumstances, and if there's no 245$f to give it an expression, things die.  I modified things to create an expression from 7-10, which has the advantage of picking up things like 19uu.  I also modified things so we're not overwriting a bulk date from 008 with an inclusive date from 245.
